File: Problem Solver/src/Problem003.cpp
#include "Problem003.h"
#include "EulerHelper.h"
#include <iostream>
#include "Primes.h" //for the isPrime() function
#include <string>

using namespace std;

Problem003::Problem003()
{
	numToFactor = 13195;  //as per Euler site.
}

Problem003::~Problem003()
{
}


void Problem003::SetNum(unsigned long long x){
	numToFactor = x;
}


void Problem003::Solve(){
    vector<unsigned long long> x = EulerHelper::factors(600851475143);

    cout << "your factors are:" << endl;
	for (long long i = 0; i <= x.size() - 1; i++){
		cout << x.at(i) << " ";
	}
	cout << endl << endl;
}

File: ProblemSolver/ProblemSolver/src/Problem003.cs
/*
 * Largest prime factor
Problem 3
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
*/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Numerics;


namespace ProblemSolver.src
{
    class Problem003 : SolvedProblems
    {
        public override void Solve(){
            List<BigInteger> x = EulerHelper.factors(BigInteger.Parse("600851475143"));

            Console.WriteLine();
            Console.WriteLine("Your factors are:");
        
            foreach(BigInteger i in x){
            Console.Write(i + " ");
            }
            Console.WriteLine();
        }
    }
}
